src/main/java/io/vertx/proton/impl/ProtonClientImpl.java

```diff
@@ -60,14 +61,43 @@ public void connect(ProtonClientOptions options, String host, int port,
   public void connect(ProtonClientOptions options, String host, int port, String username, String password,
                       Handler<AsyncResult<ProtonConnection>> handler) {
     final NetClient netClient = vertx.createNetClient(options);
-    connectNetClient(netClient, host, port, username, password, new ConnectCompletionHandler(handler), options);
+
+    NetClientConnectHandler connectHandler = new NetClientConnectHandler(netClient, host, username, password,
```

Comment:
I'm fine with this, but I would probably just have passed null for the serverName